src/core/aggregated_normalized_measures.py

```diff
@@ -0,0 +1,261 @@
+    Checker.check_metric_values(coverage, "coverage")
+
+    Checker.check_threshold(min_coverage, max_coverage, "test_coverage")
+
+    coverage, number_of_files = ems_functions.get_test_coverage(
+        data={"coverage": coverage},
+    )
+    x = coverage[coverage >= min_coverage]
+    interpretation_function_value = interpretation_function(
+        x=x,
+        min_threshold=min_coverage,
+        max_threshold=max_coverage,
+        gain_interpretation=1,
+    )
+    aggregated_and_normalized_measure = calculate_measure(
+        interpretation_function_value, number_of_files
+    )
+    return aggregated_and_normalized_measure
+
+
+def fast_test_builds(
+    data_frame, min_fast_test_time: float = 0, max_fast_test_time: float = 300000
+):
+    """
+    Calculates fast test builds (em5)
+    This function gets the dataframe metrics
+    and returns the fast test builds measure (em5).
+    """
+    test_execution_time = data_frame["test_execution_time"]
+    tests = data_frame["tests"]
+
+    Checker.check_metric_values(test_execution_time, "test_execution_time")
+    Checker.check_metric_values(tests, "tests")
+
+    Checker.check_threshold(min_fast_test_time, max_fast_test_time, "fast_test_builds")
+
+    (
+        execution_time,
+        number_of_files,
+        number_of_tests,
+    ) = ems_functions.get_fast_test_builds(
+        data={"test_execution_time": test_execution_time, "tests": tests},
+    )
+
+    # Apply threshold
+    execution_between_thresholds = execution_time[execution_time <= max_fast_test_time]
+    x = np.divide(execution_between_thresholds, number_of_tests)
+
+    interpretation_function_value = interpretation_function(
+        x=x / 100,
+        min_threshold=min_fast_test_time,
+        max_threshold=max_fast_test_time,
+        gain_interpretation=-1,
+    )
+
+    aggregated_and_normalized_measure = calculate_measure(
+        interpretation_function_value, number_of_files
+    )
+
+    return aggregated_and_normalized_measure
+
+
+def passed_tests(data_frame, min_passed_tests: float = 0, max_passed_tests: float = 1):
+    """
+    Calculates passed tests (em4)
+
+    This function gets the dataframe metrics
+    and returns the passed tests measure (em4).
+    """
+
+    tests = data_frame["tests"]  # m6 metrics
+    test_errors = data_frame["test_errors"]  # m7 metrics
+    test_failures = data_frame["test_failures"]  # m8 metrics
+
+    Checker.check_metric_values(tests, "tests")
+    Checker.check_metric_value(test_failures, "test_failures")
+    Checker.check_metric_value(test_errors, "test_errors")
+
+    Checker.check_threshold(min_passed_tests, max_passed_tests, "passed_tests")
+
+    number_of_tests, number_of_fail_tests = ems_functions.get_passed_tests(
+        data={
+            "tests": tests,
+            "test_errors": float(test_errors),
+            "test_failures": float(test_failures),
+        }
+    )
+
+    x = np.divide((number_of_tests - number_of_fail_tests), number_of_tests)
```

Comment:
Should we consider the case where the `number_of_tests` is 0?

Reply:
Quando number_of_tests=0 o resultado de x deve e será igual a zero, de qualquer maneira. Você vislumbrou alguma necessidade de tratamento @higton ?

Reply:
solved here: https://github.com/fga-eps-mds/2023-1-MeasureSoftGram-Core/commit/a2c5f46b9f675e5eb5c420bf29cd9b8f3ff09c6f